microprofile/WFLY-14277_MicroProfile_JWT_1_2_Upgrade.adoc

```diff
@@ -0,0 +1,98 @@
+= WFLY-14277 Upgrade to support  MicroProfile JWT 1.2
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The MicroProfile JWT 1.2 release is only a minor upgrade upgrading from MicroProfile 1.1 with small enhancements to APIs and supported configuration options.
+
+The full list of changes in this release can be found at:
+ * https://github.com/eclipse/microprofile-jwt-auth/releases/tag/1.2
+
+Without duplicating the list from the specification release notes, each of the updates to the specification are implemented by the SmallRye JWT project - for
+this reason it is anticipated this enhancement will be delivered as a set of component upgrades.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14277[WFLY-14277]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1596[EAP7-1596]
+* https://issues.redhat.com/browse/ELY-2068[ELY-2068]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+The component upgrades can be applied exclusively to the https://github.com/wildfly/wildfly[WildFly] project.
+
+=== Other Interested Projects
+
+The implementation for MicroProfile JWT 1.2 is contained within the https://github.com/smallrye/smallrye-jwt[SmallRye JWT] project.
+
+The https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] project contains dependencies on both 
+MicroProfile and SmallRye JWT.  However this update should not require code changes to WildFly Elytron.
+
+== Requirements
+
+=== Hard Requirements
+
+The following Maven artifact dependencies in WildFly will be updated to switch to MicroProfile JWT 1.2:
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-api
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck
+ * io.smallrye:smallrye-jwt
+ 
+Where each of these projects utilises additional dependencies this RFE will review the required dependencies and update WildFly accordingly.
+
+Similar component upgrades will also be applied to the WildFly Elytron project, however code changes are not expected there in relation
+to the upgrades.
+
+The TCK component upgrade will also require a review of the configuration of the TCK to verify the correct set of tests are executed.
+
+The SmallRye JWT project also intends to split the EE Security authentication mechanism into it's own artifact, once this is available our
+dependencies and integration within the subsystem will be updated to make use of the new artifact in addition to the current dependencies.  This
+new artifact is being delivered by the SmallRye JWT project so can be ignored as a new dependency to the application server.
```

Comment:
@darranl this is not clear to me:

- shall we wait for some further release beyond `smallrye-jwt-3.0.0-RC3` before proceeding?
- `... our dependencies and integration within the subsystem will be updated to make use of the new artifact` and `This new artifact is being delivered by the SmallRye JWT project so can be ignored as a new dependency to the application server` have a contradiction to me ... can you please explain this in more detail?

Reply:
Presently we have release candidate 3 of SmallRye JWT merged in the MP4 topic branch being used to develop the MicroProfile 4 integration - before we are complete we must move to the end unqualified release that will be produced of SmallRye JWT.

The present SmallRye JWT integration should be pretty stable now but there is always risk the migration to the unqualified release will introduce it's own changes.

The comment about the additional dependency is because one of the CI jobs against WildFly detects if a new dependency has been added to the project as that needs to be checked before it can be assumed buildable for the product releases.  The comment here is to say that any warnings about the new dependency can be ignored as it is a part of a project we already build so it is a false positive being reported.